src/Scanner.js

```diff
@@ -175,11 +175,12 @@ class DefaultScanner {
 
     /**
      * Stops scanning, switches off the mutation observer and clears all forms
+     * Keep the listener for pointerdown to scan on click if needed.
      * @param {string} reason
      * @param {any} rest
      */
-    stopScanner (reason, ...rest) {
-        this.stopped = true
+    stopScanner (reason, shouldScanOnClick = false, ...rest) {
+        this.stopped = !shouldScanOnClick
```

Comment:
Not super happy about it. I mean stopScanner should stop the scanner :/ The other thing would be to introduce a state for the scanner -> `stopped`, 'on-click`, 'running'. In my experience states like these don't end up well as you can start adding tons of those and becomes unmanageable quickly. 

The "nicest" thing I can think of now is renaming this function to `stopScannerIfneeded`. That way at least it's more indicative that it may not stop the scanner fully.

Reply:
It seems like we're moving toward the scanner having different operating modes - I don't think booleans are the correct choice for this, neither is naming 'stopped' or 'stopping'.

something like having a property called `mode` which can be 

- `scanning` <- this means it would be triggered in normal conditions
- `on-click` <- meaning it's trigged by clicks only
- 'stopped' <- meaning it should not continue 

then, instead of `stopScanner`, it can be something like `this.transition("on-click")`, or just `this.setMode('on-click')`

and instead of `this.stopped = boolean`, it would be `this.mode = <mode>`

we can strongly type the known strings to make it easier to manage.

Reply:
^ should be a very small change, but will more clearly explain the intention.

Reply:
Sounds fine with me if you think it's good! I wanted to do this at first, but thought for a quick validation, introducing new modes like this in a codebase usually sticks around and hard to refactor later. But seems like it fits the intention. Will change it.

Reply:
Added https://github.com/duckduckgo/duckduckgo-autofill/pull/617/commits/dea27ff2e80ab9546d5972c9e406c427c76e16df